app/backend/src/couchers/servicers/conversations.py

```diff
@@ -119,6 +118,24 @@ def _get_visible_admins_for_subscription(subscription):
         ]
 
 
+def _user_can_message(session, context, group_chat: GroupChat) -> bool:
+    """
+    User can message in a group chat they are part of if a chat participant exists that:
+    - Is not deleted/banned
+    - Has not been blocked by the user or is blocking the user
+    - Has not left the chat
+    """
+    return session.execute(
+        func.exists(
+            select(GroupChatSubscription)
+            .where_users_column_visible(context=context, column=GroupChatSubscription.user_id)
+            .where(GroupChatSubscription.user_id != context.user_id)
+            .where(GroupChatSubscription.group_chat_id == group_chat.conversation_id)
+            .where(GroupChatSubscription.left == None)
```

Comment:
Also would be good to add the joined check here just for completeness.

Reply:
What do you mean? What would you like me to check here?

Reply:
Actually, I think it's fine, ignore this!